bundles/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/handler/BaseModbusThingHandler.java

```diff
@@ -164,6 +178,16 @@ public ModbusSlaveEndpoint getEndpoint() {
      * @return the {@link ModbusCommunicationInterface}
      */
     private ModbusCommunicationInterface getModbus() {
+        ModbusCommunicationInterface communicationInterface = getBridgeHandler().getCommunicationInterface();
+
+        if (communicationInterface == null) {
+            throw new IllegalStateException("Failed to retrieve Modbus communication interface");
```

Comment:
According to javadoc, null communication interface is returned when initialization is incomplete == bridge is not online.

By failing fast (check bridge status) we can offer better error message to user?

Reply:
I reworked all error message to achieve easier debugging for the user.